src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -131,37 +135,51 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
                 getASMInstallsFunc = getASMInstallsFunc ?? GetAsmInstallations;
                 var installations = getASMInstallsFunc(rootFolder);
 
-                //first find the closest matches using major, minor and build version.
+                // first find the exact match or the lowest matching within same major version
                 foreach (var version in versions)
                 {
+                    Dictionary<Version, string> versionToLocationDic = new Dictionary<Version, string>();
```

Comment:
if these locations are always paths maybe it makes sense to use a `directoryInfo` or `Path` to be super explicit about what is happening. Not sure what ramifications this has though.

Reply:
I ended up using string here because ref param of DirectoryInfo below is not a good idea in general